c2/c2-client-bundle/c2-client-http/src/main/java/org/apache/nifi/c2/client/http/C2HttpClient.java

```diff
@@ -91,6 +92,45 @@ public Optional<C2HeartbeatResponse> publishHeartbeat(C2Heartbeat heartbeat) {
         return serializer.serialize(heartbeat).flatMap(this::sendHeartbeat);
     }
 
+    @Override
+    public Optional<byte[]> retrieveUpdateContent(String flowUpdateUrl) {
+        final Request.Builder requestBuilder = new Request.Builder()
+            .get()
+            .url(flowUpdateUrl);
+        final Request request = requestBuilder.build();
+
+        Optional<ResponseBody> body;
+        try (Response response = httpClientReference.get().newCall(request).execute()) {
+            int code = response.code();
+            body = Optional.ofNullable(response.body());
+
+            if (code >= HTTP_STATUS_BAD_REQUEST) {
```

Comment:
Is there a reason for checking this specific code as the minimum, as opposed to calling `Response.isSuccessful()`?